dataprep/signbibles-processing/sign-segmentation/segment_video.py

```diff
@@ -0,0 +1,237 @@
+import os
+import subprocess
+import cv2
+import mediapipe as mp
+import numpy as np
+from vidgear.gears import WriteGear
+import pympi
+import shutil
+import json
+import argparse
+import boto3
+from pathlib import Path
+import tempfile
+
+def download_from_s3(bucket_name, key, local_path):
+    """Download a file from S3 to a local path."""
+    s3_client = boto3.client('s3')
+    os.makedirs(os.path.dirname(local_path), exist_ok=True)
+    try:
+        s3_client.download_file(bucket_name, key, local_path)
+        return True
+    except Exception as e:
+        print(f"Error downloading from S3: {e}")
+        return False
+
+def process_video(input_video_path):
+    # Create segments directory if it doesn't exist
+    output_dir = os.path.join(os.path.dirname(input_video_path), '..', 'segments')
+    os.makedirs(output_dir, exist_ok=True)
+    
+    # Get base filename
+    base_filename = os.path.splitext(os.path.basename(input_video_path))[0]
+    
+    # First, run video_to_pose to get pose data
+    pose_file = os.path.join(output_dir, f"{base_filename}.pose")
+    subprocess.run(['video_to_pose', '-i', input_video_path, '-o', pose_file])
+    print("Pose extraction completed")
+    
+    # Then, run pose_to_segments to get ELAN file
+    eaf_file = os.path.join(output_dir, f"{base_filename}.eaf")
+    # Remove existing ELAN file if it exists
+    if os.path.exists(eaf_file):
+        os.remove(eaf_file)
+    subprocess.run(['pose_to_segments', '-i', pose_file, '-o', eaf_file, '--video', input_video_path])
```

Comment:
Is this `pose_to_segments` script not included?

Reply:
How do we find segments from pose information?

Reply:
The video_to_pose and pose_to_segments both are done with https://github.com/sign-language-processing/segmentation